ecs-cli/modules/cli/regcreds/regcreds_app.go

```diff
@@ -31,7 +46,184 @@ func Up(c *cli.Context) {
 	if err != nil {
 		log.Fatal("Error executing 'up': ", err)
 	}
-	log.Infof("Read creds input: %+v", credsInput) // remove after SDK calls added
 
-	//TODO: create secrets, create role, produce output
+	err = validateCredsInput(*credsInput)
+	if err != nil {
+		log.Fatal("Error executing 'up': ", err)
+	}
+
+	commandConfig := getNewCommandConfig(c)
+	updateAllowed := c.Bool(flags.UpdateExistingSecretsFlag)
+
+	smClient := secretsClient.NewSecretsManagerClient(commandConfig)
+
+	_, err = getOrCreateRegistryCredentials(credsInput.RegistryCredentials, smClient, updateAllowed)
+	if err != nil {
+		log.Fatal("Error executing 'up': ", err)
+	}
+
+	//TODO: use getOrCreate...() result to create role, produce output
+}
+
+func getOrCreateRegistryCredentials(entryMap readers.RegistryCreds, smClient secretsClient.SMClient, updateAllowed bool) (*map[string]CredsOutputEntry, error) {
+	registryResults := make(map[string]CredsOutputEntry)
+
+	for registryName, credentialEntry := range entryMap {
+		hasCredPair := hasCredPair(credentialEntry)
+		hasSecretARN := false
+		if credentialEntry.SecretManagerARN != "" {
+			hasSecretARN = true
+		}
+
+		log.Infof("Processing credentials for registry %s...", registryName)
+
+		if hasCredPair && hasSecretARN {
+			arn, err := updateOrWarnForExistingSecret(credentialEntry, updateAllowed, smClient)
```

Comment:
since the `arn` returned here is just the same as `credentialEntry.SecretManagerARN`, which is also used for setting the `hasSecretARN` var, I wonder if we could simplify this logic a bit?